src/async_client.rs

```diff
@@ -827,7 +827,7 @@ impl AsyncClient {
         let body = response.bytes().await?.as_ref().to_owned();
         let http_response = http_response.body(body).unwrap();
         let response = <Request::Response as Outgoing>::Incoming::try_from(http_response)
-            .expect("Can't convert http response into ruma response");
+            .map_err(|e| Error::LoginError(format!("{:?}", e)))?;
```

Comment:
This used to throw a normal error but ruma changed something along the way and I never figured out what error type we need to encapsulate in our own error type.

So yes this needs to be fixed, but this can't be a login error since the error can happen for any request we give.